solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java

```diff
@@ -121,25 +121,44 @@ public void close() throws IOException {
     Map<String, SolrPackageInstance> ret = new HashMap<String, SolrPackageInstance>();
     for (String packageName: packages.keySet()) {
       if (Strings.isNullOrEmpty(packageName) == false) { // There can be an empty key, storing the version here
+        if (packages.get(packageName) == null) { // This means the package was undeployed from this package before
+          continue;
+        }
         ret.put(packageName, getPackageInstance(packageName, packages.get(packageName)));
       }
     }
     return ret;
   }
 
+  private void ensureCollectionsExist(List<String> collections) {
+    try {
+      List<String> existingCollections = zkClient.getChildren("/collections", null, true);
```

Comment:
Use `ZkStateReader.COLLECTIONS_ZKNODE` constant for "/collections"

Reply:
Fixed, thanks.